python-frontend/src/main/java/org/sonar/python/tree/NumericLiteralImpl.java

```diff
@@ -67,4 +72,16 @@ public String valueAsString() {
   public List<Tree> computeChildren() {
     return Collections.singletonList(token);
   }
+
+  // https://docs.python.org/3/reference/lexical_analysis.html#numeric-literals
+  @Override
+  public InferredType type() {
+    if (valueAsStringLowerCase.contains("j")) {
+      return InferredTypes.runtimeType("complex");
+    }
+    if (valueAsString.contains(".") || valueAsStringLowerCase.contains("e")) {
+      return InferredTypes.runtimeType("float");
+    }
+    return InferredTypes.runtimeType("int");
```

Comment:
Maybe we could have constants for `InferredTypes.runtimeType("int")` and other types to avoid reinstantiating them many times.